src/fsharp/LanguageFeatures.fs

```diff
@@ -0,0 +1,90 @@
+// Copyright (c) Microsoft Corporation. All Rights Reserved. See License.txt in the project root for license information.
+
+/// Coordinating compiler operations - configuration, loading initial context, reporting errors etc.
+module internal FSharp.Compiler.Features
+
+open System
+
+//------------------------------------------------------------------------------------------------------------------
+// Language version command line switch
+//------------------------------------------------------------------------------------------------------------------
+// Add your features to this List - in code use languageVersion.SupportsFeature(LanguageFeatures.yourFeature) 
+// a return value of false means your feature is not supported by the user's language selection
+// All new language features added from now on must be protected by this.
+// Note:
+//   *  The fslang design process will require a decision about feature name and whether it is required.
+//   *  When a feature is assigned a release language, we will scrub the code of feature references and apply
+//      the Release Language version.
+
+/// LanguageFeature enumeration
+[<RequireQualifiedAccess>]
+type LanguageFeature =
+    | LanguageVersion46 = 0
```

Comment:
Is it really a language feature? Wouldn't it be better as a separate enum for language versions? It could be quite straightforward to check if a feature supported by a language version.
```
val langFeatureSupported: langVersion -> langFeature -> bool
```

Reply:
@auduchinok,

There are two audiences … compiler users are focused on language version.  Thus the langversion switch.  Feature developers are focused on their feature.  Until a feature is confirmed for a specific language version, it is assigned to the version beyond latest.  The developer merely asks if their feature is supported by the specified LanguageVersion, and invokes the feature code or the old version failure path.

When a feature is approved for an RTM language version, the code will be modified to check for that specific language version and the feature flag will disappear.   We will do this to keep the compatibility matrix from blowing our minds.  The roslyn guys tell me this is what they do too, because no one was happy with enabling individual features.